### PR TITLE
Fixed caching autodoc (failed to build). Added link to cache_file.

### PR DESCRIPTION
**Pull request by maximecb**
Added links to the `cache_file` source documentation. Fixed formatting error which prevented the `fuel.utils.cache` source docs from building.

### CODE REVIEW

**Comment by maximecb** [comment]
@nouiz Here it is

File: docs/caching.rst [hunk 1]
```diff
@@ -15,8 +15,11 @@ This variable can also be set through an environment variable as follows:
...
-    $ export FUEL_LOCAL_DATA_PATH="/home/username/my_local_cache"
+    $ export FUEL_LOCAL_DATA_PATH="/LOCAL_PATH/my_local_cache"
 
 Please note that currently, caching is only implemented in the ``H5PyDataset``.
 In order to add caching to other types of datasets, one should use the
-``cache_file`` function, which is defined in ``fuel/utils/cache.py``.
+|cache_file|_ function.
+
+.. |cache_file| replace:: ``cache_file``
+.. _cache_file: api/utils.html#fuel.utils.cache.cache_file
```
Comment (nouiz) [inline docs/caching.rst]: there is special syntax in theano doc that we use for that. Does FUEL support the same syntax?
```
    :func:`fuel.utils.cache.cache_file`
```
this remove the need to naem the html file itself.

Comment (maximecb) [reply docs/caching.rst]: Done

**Comment by maximecb** [comment]
@nouiz The docs build correctly locally, and the link to the autodoc works correctly.

**Pull request merged**